languages/elixir/exercises/concept/keyword-lists/lib/wine_cellar.ex

```diff
@@ -0,0 +1,29 @@
+defmodule WineCellar do
+  def explain_colors do
+    raise "Please implement the explain_colors/0 function"
+  end
+
+  def filter(cellar, color, opts \\ []) do
+    raise "Please implement the filter/3 function"
+  end
+
+  defp filter_by_year(_wines = [], _year), do: []
+
+  defp filter_by_year([{_, year, _} = wine | tail], year) do
+    [wine | filter_by_year(tail, year)]
+  end
+
+  defp filter_by_year([{_, _, _} | tail], year) do
+    filter_by_year(tail, year)
+  end
+
+  defp filter_by_country(_wines = [], _country), do: []
+
+  defp filter_by_country([{_, _, country} = wine | tail], country) do
+    [wine | filter_by_country(tail, country)]
+  end
+
+  defp filter_by_country([{_, _, _} | tail], country) do
+    filter_by_country(tail, country)
+  end
```

Comment:
@ErikSchierboom Is it acceptable to have an exercise that relies on a bunch of functions being already implemented in the boilerplate?

Reply:
It is definitely fine to already have code in the boilerplate. We should strive to keep the boilerplate minimal. @iHiD what do you think?

Reply:
I think I'm ok with this, because the code looks relatively simple (I don't speak Elixir so I might be wrong). I'd probably like each method to have a 1-2 line comment to explain it.

It would also be cool if this built on top of an exercise where those functions got defined by the user.

Reply:
I would agree with that these are relatively simple recursive functions.  I also think there needs to be some sort of delineation to explain them and to use them as is.

Reply:
Does this work as an explanation of the intent behind those functions? [https://github.com/exercism/v3/pull/1616/commits/d2cac1e841e7fe6a7b72d3dbb291c8127b920f2c ](https://github.com/exercism/v3/pull/1616/commits/d2cac1e841e7fe6a7b72d3dbb291c8127b920f2c)